Raven.Database/Prefetching/PrefetchingBehavior.cs

```diff
@@ -1393,7 +1393,7 @@ private static Task ObserveDiscardedTask(FutureIndexBatch source)
         {
             return source.Task.ContinueWith(task =>
             {
-                if (task.Exception != null)
+                if (task.Exception != null && !(task.Exception.InnerException is TaskCanceledException))
```

Comment:
I thought it is pretty safe to assume there are not multiple nested aggregate exceptions in this path and we can directly access `InnerException` but I haven't done in-depth analysis

Reply:
@ayende I'm still seeing the following from time to time 

```
2019-12-02 10:17:43.7516|73|Warn|Raven.Database.Prefetching.PrefetchingBehavior|Error happened on discarded future work batch
System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Raven.Database.Prefetching.PrefetchingBehavior.<>c__DisplayClass97_0.<AddFutureBatch>b__2(Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Raven.Database.Prefetching.PrefetchingBehavior.<>c__DisplayClass97_0.<AddFutureBatch>b__2(Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
---> (Inner Exception #0) System.Threading.Tasks.TaskCanceledException: A task was canceled.<---
<---

```

wonder if

```
                var isTaskCanceledException = task.Exception?.InnerException is TaskCanceledException || task.Exception?.InnerException?.InnerException is TaskCanceledException;
                if (task.Exception != null && !isTaskCanceledException)
                {
```

or logic with flattening the exception would be preferred?

Happy to send in a PR. I tried to see how I can repro this in the tests you have for the prefetch behavior but I failed to do so within reasonable time

Reply:
Added https://github.com/ravendb/ravendb/pull/9926 in case you want to proceed with this